### ISSUE TITLE
Clarification on UI orientation versus device orientation

### ISSUE THREAD

**Issue by idealopamp** [rmtmckenzie/flutter_native_device_orientation#7]
Greetings,

I'm trying to detect the orientation of the device while the UI is locked in portrait mode via     `SystemChrome.setPreferredOrientations([DeviceOrientation.portraitUp,]);`  If the phone is turned to the landscape left position, I expected this to be reflected by NativeDeviceOrientationReader.orientation, but it instead shows orientation still as portraitUp.   When I don't lock the preferred orientation, I get the expected result of landscapeLeft.  I'm assuming this means that the plugin is determining the UI orientation rather than the physical orientation of the device.  Is this the expected behavior?  If so, maybe it would be worth clarifying this is the documentation.  Also if so, would like to know if you have any plans to support actual device orientation when the UI is locked in a specific orientation, or if you would accept a PR that does support it.  

If this isn't the expected behavior, then maybe I'm doing something wrong, or maybe something got broken.

Thanks in advance!

**Comment by rmtmckenzie** [comment]
Yes, by default it currently only changes when the native window changes - I did it that way because it fit my use-case which is that I need to be able to tell when the screen rotates whether it is rotating to landscape left or landscape right (for a qr code reading app). Flutter's orientation stuff just doesn't differentiate between those two (or portait up/down but that isn't as common).

You can pass 'useSensor' to the NativeDeviceOrientationCommunicator constructor which I believe will do what you need on Android, but I haven't implemented it on iOS yet. And TBH I haven't tested that extensively (or recently) so it may not actually work properly - your mileage may vary =D.

I would definitely accept PR's for that so long as they:
1. stay under the 'useSensors' flag (or possibly a different parameter if needed) and don't affect the default
2. won't impact battery life by default (some ways of getting the sensor data ups the polling frequency which could impact battery life, which is fine if 'useSensors' is specified)
3. don't require additional permissions. If what you're trying to do does need additional permissions, I'll consider it...

Just an FYI though - depending on the APIs available on iOS you may end up tapping directly into gyroscope data, and at that point you might as well use the [Sensors plugin](https://pub.dartlang.org/packages/sensors) and use the data directly from dart.

Hope that helps =)

**Comment by desmese** [comment]
I started implementing the useSensor-parameter for the iOS-version as well. It currently works for iPhones. I still need to add some calculations to make it work correctly for an iPad as well.
I don't have a lot of experience working on iOS apps yet. So somebody may have to check it once it is done. 

While working on it I also noticed that the polling continued while the app was in the background. So I need to implement some Lifecycle management to make sure that the app only pulls sensorData when the app is in the foreground. This needs to be implemented for the Android version as well.
